proxy/src/transparency/glue.rs

```diff
@@ -15,26 +14,39 @@ use tower_service::{Service, NewService};
 use tower_h2;
 
 use ctx::transport::{Server as ServerCtx};
+use drain;
 use super::h1;
+use super::upgrade::Http11Upgrade;
+use task::{BoxSendFuture, ErasedExecutor, Executor};
 
 /// Glue between `hyper::Body` and `tower_h2::RecvBody`.
 #[derive(Debug)]
 pub enum HttpBody {
-    Http1(hyper::Body),
+    Http1 {
+        /// In HttpBody::drop, if this was an HTTP upgrade, the body is taken
+        /// to be sent on the oneshot::Sender.
```

Comment:
Woops, that comment is now slightly old. The very first version I wrote to prove it worked was by using 2 `oneshot`s, instead of the `Http11Upgrade` type. I then did the math and found that this newer version cuts atomic operations down from ~12 to ~4.

- [ ] Update comment about `oneshot`